types/tape/index.d.ts

```diff
@@ -264,6 +265,69 @@ declare namespace tape {
          * Register a callback to run after the individual test has completed. Multiple registered teardown callbacks will run in order.
          */
         teardown(callback: () => void | Promise<void>): void;
+
+        captureFn<X extends Callback>(this: Test, original: X): WrappedFn<X>;
+        // eslint-disable-next-line @definitelytyped/no-unnecessary-generics
+        capture<T extends Callback>(
+            this: Test,
+            obj: Record<PropertyKey, unknown> | unknown[],
+            method: PropertyKey,
+            implementation?: T,
+        ): WrapResults;
+        intercept(
+            obj: Record<PropertyKey, unknown> | unknown[],
+            property: PropertyKey,
+            desc?: PropertyDescriptor,
+        ): InterceptResults;
+    }
+
+    export type SyncCallback = (...args: unknown[]) => unknown;
+    export type Callback = (...args: unknown[]) => unknown | Promise<unknown>;
```

Comment:
This type may as well be `unknown`, since `unknown | Promise<unknown>` reduces to `unknown`. I am unsure why the linter didn't complain about this (which I believe it does for `any`?)

Reply:
hmm - that's fair. is there a way to represent "it can be an async function or a sync function"?

Reply:
You could say `SyncCallback | ((...args: unknown[]) => Promise<unknown>)`, but any function will still be assignable to `SyncCallback`, and anyone calling the function will still get `unknown` as the return types will be unioned.

If you don't have a specific type here, the best I think you can do is to document it.

Reply:
can't wait til i can do `unknown - Promise<unknown>` to represent a sync callback :-p

i'll just leave it as `unknown` but i'll rename it to be more descriptive